Tools/WebKitTestRunner/cg/TestInvocationCG.cpp

```diff
@@ -99,15 +100,48 @@ void computeSHA1HashStringForContext(CGContextRef bitmapContext, char hashString
         }
     }
 
-    SHA1::Digest hash;
-    sha1.computeHash(hash);
+    auto hexString = sha1.computeHexDigest();
+    
+    auto result = hexString.toStdString();
+    std::transform(result.begin(), result.end(), result.begin(), [](unsigned char c) { return toASCIILower(c); });
+    return result;
+}
 
-    hashString[0] = '\0';
-    for (size_t i = 0; i < 16; i++)
-        snprintf(hashString, 33, "%s%02x", hashString, hash[i]);
+#if PLATFORM(MAC)
+static bool operator==(const WKSize& a, const WKSize& b)
+{
+    return a.width == b.width && a.height == b.height;
 }
 
-static void dumpBitmap(CGContextRef bitmapContext, const char* checksum, WKSize imageSize, WKSize windowSize)
+static std::optional<std::string> hashForBlackImageOfSize(WKSize size)
+{
+    // Hardcode some well-known sizes for performance.
+    if (size == WKSize { 800, 600 })
+        return "6c0b5c4dc61390fd6a45eebdebf7301828ef0e56";
+    if (size == WKSize { 1600, 1200 })
+        return "11d642624de9935a659bcd042d6ff83f5a917504";
+    if (size == WKSize { 480, 360 })
+        return "feed54e9caa58b83b0c644fc037f0137828b2c76";
+
+    size_t pixelsWide = size.width;
+    size_t pixelsHigh = size.height;
+    size_t rowBytes = (4 * pixelsWide + 63) & ~63;
+
+    auto colorSpace = adoptCF(CGColorSpaceCreateDeviceRGB());
+    auto context = adoptCF(CGBitmapContextCreate(nullptr, pixelsWide, pixelsHigh, 8, rowBytes, colorSpace.get(), static_cast<uint32_t>(kCGImageAlphaPremultipliedFirst) | static_cast<uint32_t>(kCGBitmapByteOrder32Host)));
+    if (!context)
+        return { };
+
+    CGContextSetRGBFillColor(context.get(), 0, 0, 0, 1);
```

Comment:
How often do we hit the non-standard size case?

Might consider a single-variable cache of the most recent non-standard size, which would, I think improve things if the non-standard size is relatively common.

Reply:
I figured that a single WebKitTestRunner instance wouldn't benefit much from this, but maybe. I should add logging for the non-standard size case.